### PR TITLE
fix: service account and allow setting annotations

### PR DESCRIPTION
**Pull request by duboisph**
This PR fixes assignment of a ServiceAccount to the Deployment + adds the option to provide ServiceAccount annotations (eg for using with EKS IRSA).

### CODE REVIEW

**Comment by juanpicado** [comment]
@duboisph would you mind to rebase?

**Comment by duboisph** [comment]
@juanpicado no problem, rebase done :)

File: charts/verdaccio/values.yaml [hunk 1]
```diff
@@ -54,8 +54,13 @@ ingress:
 
 ## Service account
 serviceAccount:
+  # Specifies whether a service account should be created
   enabled: false
-  # name:
+  # Annotations to add to the service account
+  annotations: {}
+  # The name of the service account to use.
+  # If not set and create is true, a name is generated using the fullname template
```
Comment (todpunk) [inline charts/verdaccio/values.yaml:62]:
```suggestion
  # If not set and enabled is true, a name is generated using the Chart's fullname template
```

Comment (todpunk) [reply charts/verdaccio/values.yaml:62]: If this isn't what you were trying to convey, feel free to reject this suggestion.

**Comment by juanpicado** [comment]
Thanks guys! Merged

**Pull request merged**